### ISSUE TITLE
User Reference on Clients (Question)

### ISSUE THREAD

**Issue by mhabegger** [ebu/cpa-auth-provider#90]
According to TS 103 407 a client should use the `/token` endpoint to refresh its token with "http://https://tech.ebu.ch/cpa/1.0/client_credentials". Looking at the code it seem that this would use the UserId set on a Client object. However, neither on "association" nor on the initial "device_code" request, is the user reference set on the client object. 

Cf. https://github.com/ebu/cpa-auth-provider/blob/develop/routes/token/client-mode.js#L97

**Comment by mhabegger** [comment]
@chrisn any thoughts on this? 
In my opinion the association between client and user should happen on first token retrieval with `device_code` grant.

**Comment by chrisn** [comment]
I'll take a look through the code and get back to you.It's been a while since I last looked at this...

**Comment by chrisn** [comment]
You're right that the client-user association should be created during the `/token` request with a `device_code`. Then, when refreshing a token the `user_id` for the new token should be set to the client's `user_id` value.

I think there are a couple of bugs here. Firstly: https://github.com/ebu/cpa-auth-provider/blob/develop/test/lib/user-mode-token.js#L35 is wrong. The `user_id` field shouldn't be set against this client record. The test case is intended to check that the association is created, so this client record shouldn't have an initial `user_id` value. Second, the implementation doesn't set the client's `user_id` field when creating the token (in [user-mode.js](https://github.com/ebu/cpa-auth-provider/blob/develop/routes/token/user-mode.js)).

Good catch!
